### PR TITLE
Updated Gmsh Formula With oce

### PR DESCRIPTION
**Pull request by schoeps**
as discussed in #2785. Seem to work find. However, I see sometimes segfaults of gmsh when compiled opencascade or oce. Sorry, I can currently not debug this.

### CODE REVIEW

File: gmsh.rb [hunk 1]
```diff
@@ -7,11 +7,11 @@ def quiet_safe_system(*args)
...
   desc "Gmsh is a 3D grid generator with a build-in CAD engine."
   homepage "http://geuz.org/gmsh"
-  url "http://geuz.org/gmsh/src/gmsh-2.10.0-source.tgz"
-  sha256 "10db05a73bf7f05f6663ddb3b76045ce9decb28b36ad2e54547861254829a860"
+  url "http://geuz.org/gmsh/src/gmsh-2.10.1-source.tgz"
+  sha256 "a47f15541db038c9cb00f004d13c5648a46c3d8ebd6e0bf3b56f9274e13f505d"
 
   head "https://geuz.org/svn/gmsh/trunk", :using => GmshSvnStrategy
-  revision 1
+  revision 2
```
Comment (dpo) [inline gmsh.rb]: You can remove the revision altogether when upgrading to a new version.

File: gmsh.rb [hunk 2]
```diff
@@ -20,14 +20,26 @@ class Gmsh < Formula
...
   depends_on :mpi => [:cc, :cxx, :f90, :recommended]
   depends_on "cmake" => :build
-  depends_on "petsc" => :optional
-  depends_on "slepc" => :optional
-  depends_on "opencascade" => :recommended
-  depends_on "fltk" => :optional
-  depends_on "cairo" if build.with? "fltk"
+  depends_on "petsc"              if build.with? "petsc"
+  depends_on "slepc"              if build.with? "slepc"
+  depends_on "fltk"               if build.with? "fltk"
```
Comment (dpo) [inline gmsh.rb]: Why this change? I've always found this redundant and much prefer `:optional` dependencies.

**Comment by schoeps** [comment]
I agree more or less  :) Changes done. I will combine all commits in the end.

**Comment by dpo** [comment]
Thanks. Really minor audit issue:
```
Error: 2 problems in 1 formula
homebrew/science/gmsh:
 * 30: Trailing whitespace was found
 * 33: Trailing whitespace was found
```

**Comment by iml** [comment]
Thanks @schoeps
